docs/HowtoTDE.md

```diff
@@ -0,0 +1,219 @@
+If using third-party party software, the storage passwords need to be injected and stored as files inside AP and SM pods. NuoDB supports file per password approach explained in [using Kubernetes secrets](#using-kubernetes-secrets) and single configuration file per database approach explained in [HashiCorp Vault integration](#using-hashicorp-vault).
+Choosing the right approach depends on the capabilities and flexibility that any other 3rd party tool securing the storage passwords have.
+
+> **IMPORTANT**: NuoDB, Inc. cannot recover Transparent Data Encryption storage passwords, and the database cannot be loaded if its storage password is lost. It is the user’s responsibility to keep track of storage passwords, including historical passwords for encrypted hot copy backups.
+
+### Using Kubernetes secrets
+
+Storage passwords can be supplied using Kubernetes secret.
+One secret per database should be created in the Kubernetes namespace where NuoDB is deployed.
+It must hold the `target` and optional `historical` storage passwords for that database.
+The configured secret name is mounted inside all AP, hotcopy SM and non-hotcopy SM pods. The mount path is `/etc/nuodb/tde/<db-name>` by default and the static portion of the path can be changed using `admin.tde.storagePasswordsDir` variable.
+NuoDB Admin monitors all directories and files under `/etc/nuodb/tde` path and will configure storage passwords available in the mounted files. The following files can be used:
+
+- `/etc/nuodb/tde/<db-name>/target` = Holds the database target storage password.
+- `/etc/nuodb/tde/<db-name>/historical*` _(optional)_ = Holds a database historical storage password. Multiple files can be used to supply multiple historical passwords for each database. Historical passwords are discussed in detailin the  [Change storage passwords](#change-storage-passwords) section.
+
+Create a Kubernetes secret holding TDE passwords:
+
+```bash
+kubectl create secret generic demo-tde-secret -n nuodb \
+  --from-literal target='topSecret'
+```
+
+The secret name need to be set in the [Admin chart](../stable/admin/README.md) and [Database chart](../stable/database/README.md) during installation:
+
+```bash
+helm install -n nuodb admin stable/admin \
+  --set admin.tde.secrets.demo=demo-tde-secret
+
+helm install -n nuodb database stable/database \
+  --set admin.tde.secrets.demo=demo-tde-secret
+```
+
+### Using HashiCorp Vault
+
+[Vault Agent Sidecar Injector](https://www.vaultproject.io/docs/platform/k8s/injector) can be used to inject Vault secrets as files inside Kubernetes pods. 
+
+The scope of this document doesn't include steps on how to install and setup HashiCorp Vault. 
+
+> **NOTE**: For a quick tutorial on how to deploy Vault in Kubernetes using agent injector, see [here](https://www.hashicorp.com/blog/injecting-vault-secrets-into-kubernetes-pods-via-a-sidecar).
```

Comment:
suggest, "... using **the** agent ..."
I think it would help to replace "here" with something more descriptive. Something like,  [Injecting Vault Secrets into Kubernetes Pods]